### PR TITLE
@orta: Lets users dismiss bid confirmation view by tapping checkmark

### LINKED ISSUE #427: Make confirmation screen easier to dismiss
Saw a user tap the green circle/check mark to try and dismiss.

### PR DESCRIPTION
**Pull request by ashfurrow**
I saw a user do it at an auction once – it makes sense – it's the only saturated part on the screen, and she had been bidding already. Power user I guess :smile: 

Fixes #427.

### CODE REVIEW

**Comment by orta** [comment]
Let's get a test on this one, could be hard to remember why it exists in the future

**Comment by orta** [comment]
with that, you're cool to merge :+1:

**Comment by ashfurrow** [comment]
Cool cool.

**Comment by ashfurrow** [comment]
Trigger a gesture recognizer in tests is proving to be tricky-isn. Gonna refactor into a loading VC view model.

**Comment by ashfurrow** [comment]
Bit hairy, this. Lots  of non-obvious dependencies.

File: KioskTests/Bid Fulfillment/LoadingViewControllerTests.swift [hunk 1]
```diff
@@ -10,88 +11,117 @@ class LoadingViewControllerTests: QuickSpec {
...
+                let fulfillmentController = StubFulfillmentController()
+                let stubViewModel = StubLoadingViewModel(bidNetworkModel: BidderNetworkModel(fulfillmentController: fulfillmentController), placingBid: subject.placingBid)
+                stubViewModel.bidIsResolved = true
+                stubViewModel.isHighestBidder = true
+                subject.viewModel = stubViewModel
+
+                expect(subject).to(haveValidSnapshot())
+            }
+
+            pending("dismisses by tapping green checkmark when bidding was a success") {
```
Comment (ashfurrow) [inline KioskTests/Bid Fulfillment/LoadingViewControllerTests.swift]: TODO for tomorrow :cake:

**Comment by ashfurrow** [comment]
OK, so this PR got a lot bigger. I've done the following:

- Created a view model for `LoadingViewController` and placed all network logic within it.
- Extricated a lot of dependencies within the network models. These are now combined under the `FulfillmentController` protocol, to encourage loose coupling.
- Added tests for new classes.
- Tried to improve code commenting.

Still needs:

- [ ] Adds tests for green checkmark a.k.a "let's get a test on this one."

**Comment by ashfurrow** [comment]
While I'm at it, `PlaceBidNetworkModel`, `BidCheckingNetworkModel`, and `BidderNetworkModel` could use some tests. They currently have none :fearful:

File: Kiosk/Bid Fulfillment/FulfillmentNavigationController.swift [hunk 2]
```diff
@@ -2,15 +2,27 @@ import UIKit
 import Moya
 import ReactiveCocoa
 
-public class FulfillmentNavigationController: UINavigationController {
+// We abstract this out so that we don't have network models, etc, aware of the view controller.
```
Comment (orta) [inline Kiosk/Bid Fulfillment/FulfillmentNavigationController.swift]: :+1:

**Comment by orta** [comment]
Cool, looks good :+1:

**Pull request merged**